content/scala-cli.md

```diff
@@ -0,0 +1,150 @@
+
+The sub-commands necessary to include in the `scala` to match the functionalities of current commands:
+ - `compile`
+ - `run`
+ - `repl`
+ - `doc`
+
+On top of that, we think that the following user-facing sub-commands should also be included:
+ - `clean` - to rebuild project from start without any cached steps
+ - `setup-ide` - to control the setup of BSP for IDE integration
+ - `doctor` - to analyze if everything is installed properly
+
+We also suggest to include additional sub-commands by default: 
+ - `fmt` - to format the code using scalafmt
+ - `test` - to run tests included in the code
+ - `package` - to package the code into various package formats: JAR, "fat"  be, executable  be or even native application or docker image
+ - `shebang` - a command useful for scripting, designed to be included in the "shebang" section of scripts
+ - `export` - transform current project to sbt / Mill project using the same settings as provided. Useful to evolve prototypes into bigger projects
+
+Each of these commands expands what the current scripts offer and can be discussed separately. We can even open a dedicated SIP for each of them.
+
+Beyond that, `scala-cli` offers multiple sub-commands needed to manage itself (e.g. `update`) or its components (e.g. the Bloop server). In most cases,
+these are not user-facing, but still handy. We can elaborate in more detail on what those commands are and why we need them, if necessary.
+
+Scala CLI can also be configured with ["using directives"](https://scala-cli.virtuslab.org/docs/guides/using-directives) - a comment-based configuration syntax that should be placed at the top of Scala files. This allows for self-containing examples within one file since most of the configuration can be provided either from the command line or via using directives (command line has precedence). This is a game changer for use cases like scripting, reproduction, or within the academic scope.
+
+We have described the motivation, syntax and implementation basis in the [dedicated pre-SIP](https://contributors.scala-lang.org/t/pre-sip-using-directives/5700). Currently, we recommend to write using directives as comments, so making them part of the language specification is not necessary at this stage. Moreover, the new `scala` command could ignore using directives in the initial version, however we strongly suggest to include comment-based using directives from the start. 
+
+### Compatibility
+
+Adopting Scala CLI as the new `scala` command, as is, will change some of the behaviour of today's scripts. Some examples:
+
+- `scala repl` needs to be run to start a REPL instead of just `scala`
+- with `scala compile` and `scala doc`, some of the more obscure (or brand new) compile options need to be prefixed with `-O`
```

Comment:
We decided to curate the list of options ourselves (with keeping the list of compiler options) instead than fall back on the compiler since we have a lot of options (e.g. `--dep`) that are not forwarded, and we want to pick a types etc. in such options early especially that not all commands may call a compiler.

So this is a case where we decided that the experience of most users (especially the newcomers) is more important than 100% compatibility, especially in the really obscure cases.